modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageHttpStatsCollector.java

```diff
@@ -9,115 +9,82 @@
-        this.gcsOperationStats = gcsOperationStats;
-        this.operationPurpose = operationPurpose;
-        this.trackers = trackerFactories.stream()
-            .map(trackerFactory -> trackerFactory.apply(gcsOperationStats.getTrackedBucket()))
-            .toList();
+    private void trackRequestAndOperation(Operation operation) {
+        stats.trackRequestAndOperation(purpose, operation);
     }
 
     @Override
     public void interceptResponse(final HttpResponse response) {
-        // TODO keep track of unsuccessful requests in different entries
-        if (response.isSuccessStatusCode() == false) return;
-
-        final HttpRequest request = response.getRequest();
-        for (HttpRequestTracker tracker : trackers) {
-            if (tracker.track(request, gcsOperationStats)) {
-                return;
+        var request = response.getRequest();
+        var path = request.getUrl().getRawPath();
+        switch (request.getRequestMethod()) {
+            case "GET" -> {
+                // https://cloud.google.com/storage/docs/json_api/v1/objects/get
+                if (getObjPattern.matcher(path).matches()) {
+                    // Retrieves object metadata. When alt=media is included as a query parameter, retrieves object data.
+                    if (request.getUrl().getFirst("alt").equals("media")) {
+                        trackRequestAndOperation(Operation.GET_OBJECT);
+                    } else {
+                        trackRequestAndOperation(Operation.GET_METADATA);
+                    }
+                } else if (listObjPattern.matcher(path).matches()) {
+                    trackRequestAndOperation(Operation.LIST_OBJECTS);
+                }
+                // ignore other get requests
+            }
+            case "POST", "PUT" -> {
+                // https://cloud.google.com/storage/docs/json_api/v1/objects/insert
+                if (insertObPattern.matcher(path).matches()) {
+                    var obj = request.getUrl().getFirst("uploadType");
+                    if (obj instanceof String uploadType) {
+                        switch (uploadType) {
+                            // We dont track insert operations here, only requests. The reason is billing impact.
+                            // Any insert, including multipart or resumable parts, are counted as one operation.
+                            case "multipart" -> trackRequest(Operation.MULTIPART_UPLOAD);
+                            case "resumable" -> trackRequest(Operation.RESUMABLE_UPLOAD);
+                            default -> {
+                                // ignore "media" - Data-only upload. Upload the object data only, without any metadata.
+                            }
+                        }
+                    }
+                }
+                // ignore other post requests
+            }
+            default -> {
+                // ignore other http methods
```

Comment:
with all the methods/paths/params that we ignore, have we verified that we don't use them? might me worth logging at debug or something just to see if there's something we missed?